contracts/upgradeable_contracts/factories/HomeBridgeFactory.sol

```diff
@@ -0,0 +1,216 @@
+pragma solidity 0.4.24;
+
+import "../../IBridgeValidators.sol";
+import "../../IHomeBridge.sol";
+import "../../upgradeability/EternalStorageProxy.sol";
+import "../../upgradeability/EternalStorage.sol";
+import "../../ERC677BridgeToken.sol";
+import "../EternalOwnable.sol";
+
+contract HomeBridgeFactory is EternalStorage, EternalOwnable {
+
+  event HomeBridgeDeployed(address indexed _homeBridge, address indexed _homeValidators, address indexed _token, uint256 _blockNumber);
+
+  function getBridgeFactoryVersion() public pure returns(uint64 major, uint64 minor, uint64 patch) {
+    return (2, 2, 0);
+  }
+
+  function initialize(address _owner,
```

Comment:
We should check that the contract wasn't previously initialized, otherwise someone could override the parameters later.

We can define the following methods similar to the used on the other bridge contracts:
https://github.com/poanetwork/poa-bridge-contracts/blob/46151629adb3dd20e6fb47d4ba91b88c6d388c4f/contracts/upgradeable_contracts/BasicBridge.sol#L75-L77

https://github.com/poanetwork/poa-bridge-contracts/blob/46151629adb3dd20e6fb47d4ba91b88c6d388c4f/contracts/upgradeable_contracts/BasicBridge.sol#L79-L81

And then on `initiliaze` add 
```solidity
require(!isInitialized());
...
...
setInitialize(true);
```

The same should be implemented on `initialize` methods from `ForeignBridgeFactory` and `BridgeMapper`

Reply:
sure, will be updated